Client/DocumentServicesHelper.swift

```diff
@@ -5,54 +5,43 @@
 import JavaScriptCore
 import Shared
 import Storage
-import XCGLogger
+import NaturalLanguage
 
-private let log = Logger.browserLogger
+protocol DocumentAnalyser {
+    var name: String { get }
+    func analyse(metadata: PageMetadata) -> DerivedMetadata?
+}
+
+struct DerivedMetadata {
+    let language: String?
+}
+
+struct LanguageDetector: DocumentAnalyser {
+    let name = "NLTranslator"
+
+    func analyse(metadata: PageMetadata) -> DerivedMetadata? {
+        let text = [metadata.description, metadata.title].compactMap({$0}).joined(separator: " ")
```

Comment:
I see how this can be a problem. I'll add new metadata information about the language to PMP. That will be the best way to do this.

Reply:
I think there's three separate things to be done with page metadata parser — 

 1. detect the language that the page says its in. *Assumption*: This is likely to fail because few pages declare the natural language of the next.
 2. detect alternative languages the publisher has translated this page to.  *Assumption*: This is likely to fail because few publishers offer translations to their pages.
 3. detect some representative paragraph of text from the page that is visible to the user (i.e. not description or title). This will always succeed.